### PR TITLE
vim-patch.sh: Include upstream summary in commit message

### PR DESCRIPTION
**Pull request by jamessan**

### CODE REVIEW

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -92,8 +92,7 @@ get_vim_sources() {
 }
 
 commit_message() {
-  printf 'vim-patch:%s\n\n%s\n\n%s' "${vim_version}" \
-    "${vim_message}" "${vim_commit_url}"
+  printf '%s\n\n%s' "${vim_message}" "${vim_commit_url}"
```
Comment (justinmk) [inline scripts/vim-patch.sh]: Will runtime patches still have the `vim-patch:hash` token ?

**Comment by jamessan** [comment]
Updated commit messages look like this.

* Versioned patch
```
commit 5ef15896dc9d0950fcfd8257524b598e009e8c4f
Author: James McCoy <jamessan@jamessan.com>
Date:   Sat Dec 16 21:50:20 2017 -0500

    vim-patch:8.0.0420: text garbled when the system encoding differs from 'encoding'
    
    Problem:    When running :make the output may be in the system encoding,
                different from 'encoding'.
    Solution:   Add the 'makeencoding' option. (Ken Takata)
    
    https://github.com/vim/vim/commit/2c7292dc5bbf155fe2192d417363b8c085759cad
```

* Runtime patch
```
commit 4d63d1ce6823a101741ab78a5d642b6e67ff71ad
Author: James McCoy <jamessan@jamessan.com>
Date:   Sun Dec 17 10:18:48 2017 -0500

    vim-patch:4697399e8c80
    
    move netrw back to the previous version
    
    https://github.com/vim/vim/commit/4697399e8c805325009351a2488e3da530d0af38
```

**Comment by jamessan** [comment]
Oops, the runtime one wasn't using the updated script.

That does need to be fixed.

File: scripts/vim-patch.sh (hunk 1 above)
Comment (jamessan) [reply scripts/vim-patch.sh]: Fixed.

**Comment by jamessan** [comment]
Looks like `-s` needs to be adjusted, too.

**Comment by jamessan** [comment]
> Looks like -s needs to be adjusted, too.

Done.

File: scripts/vim-patch.sh [hunk 2]
```diff
@@ -108,22 +111,23 @@ assign_commit_details() {
...
 
   vim_commit_url="https://github.com/vim/vim/commit/${vim_commit}"
   vim_message="$(cd "${VIM_SOURCE_DIR}" \
     && git log -1 --pretty='format:%B' "${vim_commit}" \
       | sed -e 's/\(#[0-9]*\)/vim\/vim\1/g')"
-  if [[ ${strip_commit_line} == "true" ]]; then
+  if [[ ${munge_commit_line} == "true" ]]; then
     # Remove first line of commit message.
-    vim_message="$(echo "${vim_message}" | sed -e '1d')"
+    vim_message="$(echo "${vim_message}" | sed -e '1s/^patch /vim-patch:/')"
```
Comment (justinmk) [inline scripts/vim-patch.sh]: Is this dependent on Bram never changing the message format? In particular, we never want whitespace after `vim-patch:` token.

Comment (jamessan) [reply scripts/vim-patch.sh]: He has a script that handles this.  It [looks like](https://gist.github.com/jamessan/0c8e3770f83c7a3d9eaa204e55eb4245) we're good.  The last 1300 patches or so don't violate that format.

**Review by justinmk — APPROVED** [review]

**Pull request merged**